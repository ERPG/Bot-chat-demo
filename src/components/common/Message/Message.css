@import '../../Chat/InputWrapper/InputWrapper.css';

.landbot-message {
  margin-top: 0;
  padding-top: 0;
}

.landbot-message + .landbot-message {
  margin-top: 12px;
  padding-top: 0;
  border-top: 0;
}

.landbot-message[data-author="bot"] + .landbot-message[data-author="bot"],
.landbot-message[data-author="user"] + .landbot-message[data-author="user"] {
  margin-top: 3px;
}

.landbot-message[data-author="user"] {
  justify-content: flex-end;
}

.landbot-message-avatar {
  display: none;
}

.landbot-message-content {
  color: var(--light);
  padding: 8px 15px;
  border-radius: 18px;
  display: inline-block;
  flex-grow: 0;
}

.landbot-message[data-author="bot"] .landbot-message-content {
  background-color: var(--dark-blue-2);
  border-bottom-left-radius: 2px;
}

.landbot-message[data-author="user"] .landbot-message-content {
  background-color: var(--orange);
  border-bottom-right-radius: 2px;
}