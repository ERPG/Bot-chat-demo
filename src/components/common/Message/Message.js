import React from 'react';

import './Message.css';

/**
 * Represents a Message.
 * @constructor
 * @param {string} key - message key.
 * @param {string} author - The author of the message that can be Bot or User.
 * @param {string} text - The text message.
 * There are more props in this element, for the purpose of this test we will use this few.
 */

const Message = ({key, author, text}) => {
  return (
    <article
      key={key}
      data-author={author}
      data-landbot="message"
      className="media landbot-message"
    >
      <figure className="media-left landbot-message-avatar">
        <p className="image is-64x64">
          <img className="is-rounded" src="http://i.pravatar.cc/100" alt="" />
        </p>
      </figure>
      <div className="media-content landbot-message-content">
        <div className="content">
          <p>{text}</p>
        </div>
      </div>
    </article>
  )
}

export default Message;