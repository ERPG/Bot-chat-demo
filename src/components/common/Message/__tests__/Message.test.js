import React from 'react';
import { render } from '@testing-library/react';
import { customRender } from '../../../../../__tests__/__utils__/testUtils';
import Message from '../Message';

describe('Message component', () => {
  const messagesSelector = 'message';

  const messagesProps = {
    key: '',
    author:'bot',
    text:'hola'
  }
  beforeEach(() => {
    jest.clearAllMocks();
  })

  it('Should render component without errors', () => {
    const { container } = render(<Message {...messagesProps}/>);
    expect(container).toMatchSnapshot();
  });

  it('Should message be in the document', async() => {
    const { getByDataLb } = customRender(<Message {...messagesProps} />);
    const message = getByDataLb(messagesSelector)
    expect(message).toBeInTheDocument();
  });
});