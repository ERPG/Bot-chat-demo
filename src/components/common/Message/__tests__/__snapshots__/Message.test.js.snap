// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component Should render component without errors 1`] = `
<div>
  <article
    class="media landbot-message"
    data-author="bot"
    data-landbot="message"
  >
    <figure
      class="media-left landbot-message-avatar"
    >
      <p
        class="image is-64x64"
      >
        <img
          alt=""
          class="is-rounded"
          src="http://i.pravatar.cc/100"
        />
      </p>
    </figure>
    <div
      class="media-content landbot-message-content"
    >
      <div
        class="content"
      >
        <p>
          hola
        </p>
      </div>
    </div>
  </article>
</div>
`;
